src/EFCore/Metadata/Internal/EntityType.cs

```
@@ -2541,36 +2541,50 @@ public override void OnTypeMemberIgnored(string name)
                 {
                     if (!properties.TryGetValue(memberInfo.GetSimpleMemberName(), out var propertyBase))
                     {
+                        if (memberInfo is PropertyInfo propertyInfo
+                            && propertyInfo.IsIndexerProperty())
```

We should process indexers after other properties to avoid these issues:

- Overriding a value from a non indexer property
- Trying the get a value from the indexer that could be obtained from a normal property
- Trying to get a value from the indexer that's not supplied for an optional property (we'll probably need to swallow an exception)

With this in mind we could only use the indexer if the property is indexed
Also add tests for the above cases

---

Actually we should throw if and indexed property name matches a member on the CLR type

---

I would be happy with validation to avoid the clash scenario. We can add support later if customers requests for it.

So for initial points, how many remains valid?
3rd one I will add try/catch.

---

So to handle everything you would need to

- Only use PropertyInfo defined in the model for non-anonymous types
- Add try/catch for optional indexer properties
- Validate indexer property doesn't clash

---

added test for validation.